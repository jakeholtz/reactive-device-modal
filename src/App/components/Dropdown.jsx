import React, { Component } from "react";

export default class Dropdown extends Component {
  constructor(props) {
    super(props);
    this.state = {isOpen: false};
    this.handleClick = this.handleClick.bind(this);
    this.toggleDropdown = this.toggleDropdown.bind(this);

    /* Set initial inputValue as first value in options list */
    if (props.options && props.options.length) this.setInitialState.call(this, props.options[0]);
  }

  toggleDropdown() {
    const { isOpen } = this.state;
    this.setState({isOpen: !isOpen});
  }

  setInitialState(inputValue) {
    const { isOpen } = this.state; 
    this.state = { inputValue, isOpen };
    this.props.updateInput(this.props.name, inputValue);
  }

  handleClick(inputValue) {
    this.setState({inputValue});
    this.props.updateInput(this.props.name, inputValue);
  }

  render() {
    const { options } = this.props;

    return (
    <div className="dropdown" onClick={this.toggleDropdown}>
      <div className="dropdown-select dropdown-trigger">
        <span>{ this.state.inputValue }</span>
        {  this.state.isOpen ?
           <span className="up-arrow">&#x25B4;</span> :
           <span className="down-arrow">&#x25BE;</span>
        }
      </div>
      { this.state.isOpen ? 
          (<div className="dropdown-content">
            { options.map((opt, index) => <div className={`dropdown-item ${opt === this.state.inputValue ? 'selected-item' : ''}`}
                                               onClick={() => this.handleClick(opt)} key={index}>{opt}
                                            <span className="dropdown-check">&#10003;</span>
                                          </div> ) }  
          </div>) : ''
      }
    </div>
    );
  }
}