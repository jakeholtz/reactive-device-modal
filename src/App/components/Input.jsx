import React, { Component } from "react";

export default class Input extends Component {
  constructor(props) {
    super(props);
    this.state = { inputValue: '' };
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange(event) {
    this.setState({inputValue: event.target.value});
    this.props.updateInput(this.props.name, event.target.value);
  }

  render() {
    const { name, type, placeholder } = this.props;

    return (
        <form className="form-container">
          <input className="form-input"
                 name={name}
                 type={type}
                 placeholder={placeholder}
                 value={this.state.inputValue}
                 onChange={this.handleChange} />
        </form>
      );
  }
}