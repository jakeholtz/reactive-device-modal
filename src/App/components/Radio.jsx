import React, { Component } from "react";

export default class Radio extends Component {
  constructor(props) {
    super(props);
    this.state = { inputValue: '' };
    this.handleChange = this.handleChange.bind(this);

    /* Set initial inputValue as first value in options list */
    if (props.options && props.options.length) this.setInitialState.call(this, props.options[0]);
  }

  setInitialState(inputValue) {
    this.state = { inputValue };
    this.props.updateInput(this.props.name, inputValue);
  }

  handleChange(event) {
    this.setState({inputValue: event.target.value});
    this.props.updateInput(this.props.name, event.target.value);
  }

  render() {
    const { options } = this.props;

    return (
      <form className="form-radio-container" >
        <div className="radio-container">
          {
            options.map((option, index) => {
              const identifier = `${this.props.name}${index + 1}`;
              return (
                <div className="radio-option" key={identifier} >
                  <input className="radio-input" name={this.props.name} type="radio" value={option}
                         checked={this.state.inputValue ? this.state.inputValue === option : !index}
                         onChange={ this.handleChange } />
                  <label className="radio-label" htmlFor={identifier}>{option}</label>
                </div>
              );
            })
          }
        </div>
      </form>
    );
  }
}