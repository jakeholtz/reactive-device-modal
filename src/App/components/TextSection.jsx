import React, { Component } from "react";
import styles from "../App.less";

export default class TextSection extends Component {
  render() {
    const { title, text } = this.props;

    return (
      <section className="section" >
        <h5 className="section-title"> { title } </h5>
        <p className="section-text"> { text } </p>
        <div className="section-content"></div>
      </section>
    );
  }
}