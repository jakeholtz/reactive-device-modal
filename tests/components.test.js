import React from 'react';
import { shallow } from 'enzyme';
import Input from '../src/App/components/Input.jsx';
import Radio from '../src/App/components/Radio.jsx';
import Dropdown from '../src/App/components/Radio.jsx';
import TextSection from '../src/App/components/TextSection.jsx';

describe('Components should render correctly in "debug" mode', () => {
  it('<Dropdown /> component should render', () => {
    const dropdownComponent = shallow(<Dropdown options={['A', 'B', 'C']} updateInput={() => {}} debug />);
    expect(dropdownComponent).toMatchSnapshot();
  });

  it('<Input /> component should render', () => {
    const inputComponent = shallow(<Input debug />);
    expect(inputComponent).toMatchSnapshot();
  });

  it('<Radio /> component should render', () => {
    const radioComponent = shallow(<Radio options={['A', 'B', 'C']} updateInput={() => {}} debug />);
    expect(radioComponent).toMatchSnapshot();
  });

  it('<TextSection /> component should render', () => {
    const textSectionComponent = shallow(<TextSection debug />);
    expect(textSectionComponent).toMatchSnapshot();
  });
});

describe('Input Component', () => {
  const inputComponent = <Radio placeholder="email" onChange={() => {}} debug />;

  it('Should include placeholder in props', () => {
    expect(inputComponent.props.placeholder).toEqual('email');
  });
  it('Should have method to handle input change', () => {
    expect(typeof inputComponent.props.onChange).toEqual('function');
  });
});


describe('Radio Component', () => {
  const options = ['A', 'B', 'C'];
  const radioComponent = <Radio options={options} updateInput={() => {}} debug />;

  it('Should include all options passed in', () => {
    expect(radioComponent.props.options.length).toEqual(3);
  });
  it('Should have method to update user input', () => {
    expect(typeof radioComponent.props.updateInput).toEqual('function');
  });
});

describe('TextSection Component', () => {
  const expectedTitle = 'Test Title';
  const expectedText = 'Test Text';
  const textSection = <TextSection title={expectedTitle} text={expectedText} debug />;

  it('Should render title correctly', () => {
    expect(textSection.props.title).toEqual(expectedTitle);
  });
  it('Should render text correctly', () => {
    expect(textSection.props.text).toEqual(expectedText);
  });
});