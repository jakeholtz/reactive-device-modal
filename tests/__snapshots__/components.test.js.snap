// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components should render correctly in "debug" mode <Dropdown /> component should render 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly in "debug" mode <Input /> component should render 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly in "debug" mode <Radio /> component should render 1`] = `ShallowWrapper {}`;

exports[`Components should render correctly in "debug" mode <TextSection /> component should render 1`] = `ShallowWrapper {}`;
